ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -1095,45 +1130,53 @@ data MultiNodePruningScript req = MultiNodePruningScript
   deriving (Show)
 
 -- | To generate well-formed scripts we need to keep track of what nodes are started and what
---   connections they've made.
+--   connections they've made. We also need to track the delay each peer action took so we
+--   can make sure, e.g. that we don't have:
+--
+--   OutboundConnection 0.1s A -> CloseOutboundConnection 0s A
+--
+--   We should consider the delay of creating outbound connection to A:
+--
+--   OutboundConnection 0.1s A -> CloseOutboundConnection 0.1s A
 --
 --   Note: this does not track failures, e.g. `requestOutboundConnection` when there's
 --   already a `Unidirectional` inbound connection (i.e. a `ForbiddenOperation`).
 --
-data ScriptState peerAddr = ScriptState { startedClients      :: [peerAddr]
-                                        , startedServers      :: [peerAddr]
-                                        , clientConnections   :: [peerAddr]
-                                        , inboundConnections  :: [peerAddr]
-                                        , outboundConnections :: [peerAddr] }
+data ScriptState peerAddr = ScriptState { startedClients      :: [(peerAddr, DiffTime)]
+                                        , startedServers      :: [(peerAddr, DiffTime)]
+                                        , clientConnections   :: [(peerAddr, DiffTime)]
+                                        , inboundConnections  :: [(peerAddr, DiffTime)]
+                                        , outboundConnections :: [(peerAddr, DiffTime)]
```

Comment:
Where do we use the `DiffTime`?

Reply:
Good point, I don't know why this was added by me

Reply:
Oh I remember, it is to make sure we don't generate the following:

```
(...) OutboundConnection 0.1s (TestAddr {unTestAddr = TestAddress (39)}),
(...),CloseOutboundConnection 0s (TestAddr {unTestAddr = TestAddress (39)}
```

Reply:
Why is this a problem?

Reply:
You answered that in the issue:

> In this particular test we issue an inbound connection that makes progress and then we issue a close connection before the outbound connection has been established.

Reply:
Can we in that case block until the connection is established? I don't like using timeouts for synchronisation.